pkg/controller/replicaset/replica_set.go

```diff
@@ -681,14 +725,57 @@ func slowStartBatch(count int, initialBatchSize int, fn func() error) (int, erro
 	return successes, nil
 }
 
-func getPodsToDelete(filteredPods []*v1.Pod, diff int) []*v1.Pod {
+// getIndirectlyRelatedPods returns all pods that are owned by any ReplicaSet
```

Comment:
I would expect a table test for this method

Reply:
I made `TestRelatedPodsLookup` table-driven—is that what you had in mind?  https://github.com/kubernetes/kubernetes/pull/80004/commits/de3f4955c09a7159974ccb8479e8edbd87c009e9#diff-16e152a1acd1373ada706d8cb4c099f6R507-R535